File: 27.remove-element.cpp
/*
 * @lc app=leetcode id=27 lang=cpp
 *
 * [27] Remove Element
 */

// @lc code=start
#include <cstdint>
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        uint32_t fast = 0;
        uint32_t slow = 0;
        const uint32_t t = nums.size();
        while (fast < t) {
            while (fast < t && nums[fast] == val) {
                fast++;
            }
            if (fast < t) {
                nums[slow] = nums[fast];
                slow++;
                fast++;
            }
        }
        return slow;
    }
};
// @lc code=end

File: 27.remove-element.py
#
# @lc app=leetcode id=27 lang=python3
#
# [27] Remove Element
#

# @lc code=start
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        fast = 0
        slow = 0
        while fast < len(nums):
            while fast < len(nums) and nums[fast] == val:
                fast += 1
            if fast < len(nums):
                nums[slow] = nums[fast]
                slow += 1
                fast += 1
        return slow

# @lc code=end

